### PR TITLE
feat: add a docs action

### PR DESCRIPTION
**Pull request by kneal**
This PR adds the ability to generate documentation to live within this repository. The generation is a built-in feature of [urfave/cli](https://github.com/urfave/cli). For details on their implementation, you can see https://github.com/urfave/cli/pull/830

There is a Lorem ipsum example available that shows the implementation. 
https://github.com/jribe/cli/wiki/Markdown-doc-example

The plan will be to reference the new `DOCS.md` file in this repository directly on the Vela documentation site. This will make it so all docs on the site are sourced from the commands directly and generated then propagated to our site.

### CODE REVIEW

**Comment by kneal** [comment]
Per feedback making the command hidden: https://github.com/go-vela/cli/pull/191/commits/71ee65b86c07fe3aeff2d3c8a6334e7c25389a16

```release/darwin/amd64/vela gn
NAME:
   vela generate - Generate resources for Vela via subcommands

USAGE:
   vela generate command [command options] [arguments...]

DESCRIPTION:
   Use this command to generate resources for Vela.

COMMANDS:
   completion  Generate a shell auto completion script
   config      Generate the config file used in the CLI
   pipeline    Generate a valid Vela pipeline
   help, h     Shows a list of commands or help for one command```

File: action/docs/generate.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+		// output the markdown in stdout format
+		//
+		// https://pkg.go.dev/github.com/go-vela/cli/internal/output?tab=doc#Stdout
+		return output.Stdout(markdown)
+	case c.Man:
+		logrus.Tracef("creating man pages documentation")
+
+		// generate the documentation from the application configuration
+		man, err := a.ToMarkdown()
```
Comment (davidvader) [inline action/docs/generate.go:39]: .toMan

File: DOCS.md [hunk 2]
```diff
@@ -0,0 +1,710 @@
...
+
+**--api.token, -t**="": token used for communication with the Vela server
+
+**--api.token.access, --at**="": access token used for communication with the Vela server
+
+**--api.token.refresh, --rt**="": refresh access token used for communication with the Vela server
+
+**--api.version, --av**="": API version for communication with the Vela server (default: v1)
+
+**--config, -c**="": path to Vela configuration file (default: /Users/Z001NR1/.vela/config.yml)
```
Comment (jbrockopp) [inline DOCS.md:37]: I believe this will need to be updated.. the default path looks inaccurate

File: action/docs_generate.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+DOCUMENTATION:
+
+  https://go-vela.github.io/docs/cli/docs/generate/
+`, cli.CommandHelpTemplate),
+}
+
+// helper function to capture the provided
+// input and create the object used to
+// produce the config file.
```
Comment (davidvader) [inline action/docs_generate.go:56]: typo?
```
// produce the config file.

```

**Review by jbrockopp — CHANGES_REQUESTED** [review]

**Review by matt-fevold — APPROVED** [review]
looks like all feedback was addressed. LGTM 🐕

**Review by davidvader — APPROVED** [review]
ty for making those updates, LGTM

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**